### PR TITLE
Calling ERT via the communication console.

### PR DESCRIPTION
**Pull request by VigersRay**
## About the PR

Made it possible to call the ERT through the communication console with the verification of certain conditions. Perhaps this needs further improvement, in particular, the issue of calling the ERT during nuclear operatives and the ability to turn it off through swaras, as well as the ability to cancel the ERT called by the admins, is open.

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

image
image

https://github.com/space-wizards/space-station-14/assets/60344369/ad2b0bd5-1f0f-4324-9506-dcb273eaafa3

**Changelog**

:cl: VigersRay
- add: Nanotransen made it possible for stations to request ERT via the communications console.

### CODE REVIEW

**Comment by Dynexust** [comment]
Good idea, however I feel as if the admins should be able to approve manually whether to send an ERT or not.

**Comment by Maksim22848** [comment]
the idea is really good, but there will be a big problem that the players will just shitspawn them

**Comment by VigersRay** [comment]
> the idea is really good, but there will be a big problem that the players will just shitspawn them

There are requirements for each type and fallback between calls, so I don't think that's a problem.

**Comment by ficcialfaint** [comment]
why’s it 14k lines

**Comment by VigersRay** [comment]
> why’s it 14k lines

shuttles...

**Comment by GeneralGaws** [comment]
> why’s it 14k lines 

 Shuttles for every type of ert squad, cburn shuttle, med shuttle etc

**Comment by ficcialfaint** [comment]
i see mb

**Comment by noverd** [comment]
Good.

File: Resources/Prototypes/Roles/Jobs/Fun/emergencyresponseteam.yml [hunk 1]
```diff
@@ -222,3 +222,16 @@
     ears: ClothingHeadsetAltCommand
     belt: ClothingBeltJanitorFilled
     pocket1: Flare
+
+
+- type: startingGear
+  id: ERTClownGear
+  equipment:
+    jumpsuit: ClothingUniformJumpsuitClown
+    back: ClothingBackpackClownFilled
```
Comment (PuroSlavKing) [inline Resources/Prototypes/Roles/Jobs/Fun/emergencyresponseteam.yml:231]: "ClothingBackpackERTClown"?

**Comment by Just-a-Unity-Dev** [comment]
this should be a cvar i can see some servers wanting it while others not

**Comment by Just-a-Unity-Dev** [comment]
also i find ERT to be unnecessary cause command will literally just call ERT for everything don't tell me they won't

File: Resources/Prototypes/Roles/Jobs/Fun/emergencyresponseteam.yml (hunk 1 above)
Comment (VigersRay) [reply Resources/Prototypes/Roles/Jobs/Fun/emergencyresponseteam.yml:231]: is real?

**Comment by VigersRay** [comment]
> this should be a cvar i can see some servers wanting it while others not

In general, it is easy to make a cvar, but the call button itself will remain in both cases.

**Comment by VigersRay** [comment]
> also i find ERT to be unnecessary cause command will literally just call ERT for everything don't tell me they won't

In most cases, ERT will be automatically rejected.

**Comment by VigersRay** [comment]
> Yeah a system, where an admin should always be in control here I think is best. Basically you could put a request in at the console, and the admin would have to approve it. If not approved, you could get a centcomm message like, "Sorry all our Ert teams are busy at the moment leave a message at the beep."

Actually the whole idea is to allow the crew to receive ERT without administrators. Perhaps there are always admins on the servers of the wizards, but on certain servers there is a problem with this, the admins are either too lazy to spawn the return shuttle and roles every time, or they simply do not exist on the server in most cases. I just heard that the call to ERT in cc13 also happens automatically without the participation of admins.

**Comment by KaiShibaa** [comment]
Personally, ERT should be admin only. People are going to end up abusing this for every single little thing. ERT should also not be an answer to everything.

 I don't think calling admins lazy is a good idea either.

**Comment by CloneTrooperSimp** [comment]
Interesting idea.
We hope it won't get abused by players, but that is... expecting too much I'd say...

**Comment by Nairodian** [comment]
Admins _should_ probably have some button to prevent calling it, in case of events and such. But if there's a system in place to require two or more command votes, admin denial and accepting may not be required.

**Comment by Geekyhobo** [comment]
As a non 13 player I didn't think about the head vote thing, that I think is the winner, especially with the increased amount of hours to play command roles, I believe that the heads can make the right call.

**Comment by VigersRay** [comment]
> 

On the server where I have already uploaded this, oddly enough they do not abuse it, even during the attack of nuclear operatives, no one called the ERT having only 5 sats in the state. I think this only applies to LRP servers. While again, there are fixed requirements for sending an ERT , these can be increased by making the ability to call an ERT via the console very rare, such as 80% dead crew, or 80% zombies on a station.

**Comment by VigersRay** [comment]
> Personally, ERT should be admin only. People are going to end up abusing this for every single little thing. ERT should also not be an answer to everything.
> 
> I don't think calling admins lazy is a good idea either.

On the server where I have already uploaded this, oddly enough they do not abuse it, even during the attack of nuclear operatives, no one called the ERT having only 5 sats in the state. I think this only applies to LRP servers. While again, there are fixed requirements for sending an ERT , these can be increased by making the ability to call an ERT via the console very rare, such as 80% dead crew, or 80% zombies on a station.

**Comment by lzk228** [comment]
player abusing, player gets banned, isnt it?

**Comment by metalgearsloth** [comment]
We ended up disabling early launches entirely as our playerbase abused it constantly so I would put money on this being abused.

**Comment by VigersRay** [comment]
> We ended up disabling early launches entirely as our playerbase abused it constantly so I would put money on this being abused.

Perhaps if there was the same system for early launch as in expeditions (checking if all the utilizers have returned), this problem would not exist. It would be possible to check conditionally if 80-90% of the LIVE crew came on board, then an early launch can be made.

**Comment by Peptide90** [comment]
> player abusing, player gets banned, isnt it?

Relying on admin intervention is the last step in the process. If you can engineer it out do that first.

I'm used to SS13 where you need two head swipes in like 5 seconds. Same thing exists for shuttle calling there too. Also +1 for allowing admins to block all requests.

**Comment by deltanedas** [comment]
if admins are online this should be a thing that requires admin approval to spawn, with an option in adminbus or something to disable entirely since average lizard shitter will call it after stubbing his toe and might mess with events as previously stated

with no admins the coded checks should help but ultimately it would be a burden to deal with reports of abusing it

**Comment by JoeHammad1844** [comment]
what are the requirements for ERT to be called? and please send images of all the shuttles

**Comment by metalgearsloth** [comment]
Swipes need coding

**Review by AJCM-git — CHANGES_REQUESTED** [review]
Heads swipes as described by PJB are necessary

**Pull request closed without merging**